rlog_test.go

```diff
@@ -0,0 +1,241 @@
+// All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+package rlog
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path"
+	"runtime"
+	"testing"
+	"time"
+)
+
+var logfile string
+var removeLogfile bool = false
+var fixedLogfileName bool = true
+
+// setup is called at the start of each test and prepares a new log file. It
+// also returns a new configuration, which can be used by this test.
+func setup() rlogConfig {
+	if fixedLogfileName {
+		logfile = "/tmp/rlog-test.log"
+	} else {
+		logfile = fmt.Sprintf("/tmp/rlog-test-%d.log", time.Now().UnixNano())
+	}
+	os.Remove(logfile)
+
+	return rlogConfig{
+		logLevel:       "",
+		traceLevel:     "",
+		logTimeFormat:  "",
+		logFile:        logfile,
+		logStream:      "NONE",
+		logTimeDate:    false,
+		showCallerInfo: false,
+	}
+}
+
+// cleanup is called at the end of each test.
+func cleanup() {
+	if removeLogfile {
+		os.Remove(logfile)
+	}
+}
+
+// fileMatch compares entries in the logfile with expected entries provided as
+// a list of strings (one for each line)
+func fileMatch(t *testing.T, checkLines []string, timeLayout string) {
+	file, err := os.Open(logfile)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	i := 0
+	for scanner.Scan() {
+		line := scanner.Text()
+		if timeLayout != "" {
+			dateTime := line[:len(timeLayout)]
+			line = line[len(timeLayout)+1:]
+			_, err := time.Parse(timeLayout, dateTime)
+			if err != nil {
+				t.Fatal(err)
+				t.Fatalf("Incorrect date/time format.\nSHOULD: %s\nIS:     %s\n", timeLayout, dateTime)
+			}
+		}
+		if i >= len(checkLines) {
+			t.Fatal("Not enough lines provided in checkLines.")
+		}
+		if line != checkLines[i] {
+			t.Fatalf("Log line %d does not match check line.\nSHOULD: %s\nIS:     %s\n", i, checkLines[i], line)
+		}
+		i++
+	}
+	if len(checkLines) > i {
+		t.Fatalf("Only %d of %d checklines found in output file.", i, len(checkLines))
+	}
+	if i == 0 {
+		t.Fatal("No input scanned")
+	}
+}
+
+func TestLogLevels(t *testing.T) {
+	conf := setup()
+	defer cleanup()
+
+	conf.logLevel = "DEBUG"
+	Initialize(conf)
+
+	Debug("Test Debug")
+	Info("Test Info")
+	Warn("Test Warning")
+	Error("Test Error")
+	Critical("Test Critical")
+
+	checkLines := []string{
+		"DEBUG    : Test Debug",
+		"INFO     : Test Info",
+		"WARN     : Test Warning",
+		"ERROR    : Test Error",
+		"CRITICAL : Test Critical",
+	}
+	fileMatch(t, checkLines, "")
```

Comment:
Ah, that's how I can get output during tests? Drove me nuts that I couldn't just print things! :-)